File: platform/sdk/python/examples/t3_05_list_uploads.py
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))
from mox_sdk_cloud import CloudClient

def main():
    client = CloudClient()
    names = ["alpha", "beta", "gamma", "delta"]
    for i, n in enumerate(names):
        r0 = client.createMultipartUpload("lb", f"file/{n}.bin")
        if not r0["ok"]:
            raise RuntimeError(f"create failed for {n}")
        uid = r0["upload_id"]
        if i % 2 == 0:
            client.uploadPart("lb", f"file/{n}.bin", uid, 1, bytes(64))
    lst = client.listMultipartUploads()
    if lst["count"] != 4:
        raise RuntimeError(f"expected 4 uploads got {lst['count']}")
    with_parts = sum(1 for m in lst["uploads"] if m["parts_count"] > 0)
    if with_parts != 2:
        raise RuntimeError(f"expected 2 with parts got {with_parts}")
    print(f"XJ-OK: t3_05_list_uploads total={lst['count']} with_parts={with_parts}")

if __name__ == "__main__":
    main()

File: platform/sdk/nodejs/examples/t3-05-list-uploads.js
const { CloudClient } = require("../mox-sdk-cloud");

function main() {
  const client = new CloudClient();
  const names = ["alpha", "beta", "gamma", "delta"];
  names.forEach((n, i) => {
    const r0 = client.createMultipartUpload("lb", "file/" + n + ".bin");
    if (!r0.ok) throw new Error("create failed for " + n);
    const uid = r0.upload_id;
    if (i % 2 === 0) {
      client.uploadPart("lb", "file/" + n + ".bin", uid, 1, Buffer.alloc(64, 0));
    }
  });
  const list = client.listMultipartUploads();
  if (list.count !== 4) throw new Error("expected 4 uploads, got " + list.count);
  const withParts = list.uploads.filter(m => m.parts_count > 0).length;
  if (withParts !== 2) throw new Error("expected 2 uploads with parts, got " + withParts);
  console.log("XJ-OK: t3-05-list-uploads total=" + list.count + " with_parts=" + withParts);
}

if (require.main === module) main();
module.exports = main;
